src/qbe_ir_generator.cpp

```diff
@@ -205,11 +205,34 @@ void QbeIrGenerator::Visit(const ArrDeclNode& arr_decl) {
   }
 }
 
-void QbeIrGenerator::Visit(const RecordDeclNode& struct_def) {}
+void QbeIrGenerator::Visit(const RecordDeclNode& record_decl) {
+  /* Do nothing because this node only declares a type. */
+}
+
+void QbeIrGenerator::Visit(const FieldNode& field) {
+  /* Do nothing because this node only declares a member type in a record. */
+}
 
-void QbeIrGenerator::Visit(const FieldNode& field) {}
+void QbeIrGenerator::Visit(const RecordVarDeclNode& record_var_decl) {
+  const auto base_addr = NextLocalNum();
+  // TODO: support different data types. We have `int` type for now.
+  WriteInstr_("{} =l alloc4 {}", FuncScopeTemp{base_addr},
+              record_var_decl.type->size());
+  id_to_num[record_var_decl.id] = base_addr;
 
-void QbeIrGenerator::Visit(const RecordVarDeclNode& struct_def) {}
+  for (auto i = std::size_t{0}, e = record_var_decl.inits.size(); i < e; ++i) {
```

Comment:
In the case of excess elements when initializing a union, for example, `union shape s = {3, 4, 5};`, it seems this loop iterates 3 times and stores the elements at offsets 0, 4, and 8. This behavior doesn't seem correct. However, I haven't run the code to verify this. Could you please clarify how this loop handles such cases?

Reply:
> In the case of excess elements when initializing a union, for example, `union shape s = {3, 4, 5};`, it seems this loop iterates 3 times and stores the elements at offsets 0, 4, and 8. This behavior doesn't seem correct. However, I haven't run the code to verify this. Could you please clarify how this loop handles such cases?

I think you're right. I'll need to handle union as a special case, or else compiler will store garbage value. 🗑️

Reply:
Or maybe you stop iterating when the offset is greater than the size of the record?

Reply:
> Or maybe you stop iterating when the offset is greater than the size of the record?

Let me know what you think about mine approach. Since we only update the first member for multiple initializers, abort union code generation if the loop iterates more than once.

Reply:
I just find out that `struct` also allows excessive elements. Let's not make `union` special. :sweat_smile: